redaxo/src/addons/media_manager/lib/managed_media.php

```diff
@@ -296,7 +296,7 @@ public function getSource()
             return $this->getImageSource();
         }
 
-        return rex_file::get($this->sourcePath);
+        return rex_file::get($this->sourcePath, '');
```

Comment:
So ganz glücklich bin ich mit der Lösung nicht, überall einfach leeren String als Default zu nehmen.
Kaschieren wir so nicht das Problem, falls Datei nicht vorhanden, oder nicht gelesen werden kann, noch mehr? Wenn wir dann einfach mit leerem String weiterarbeiten?

Ich schlage eher sowas vor wie `rex_file::require`, was dann eine Exception wirft, wenn die Datei nicht gelesen werden kann.
`require` haben wir ja ähnlich in anderen Klassen auch schon eingeführt.
Wobei es hier in rex_file missverständlich sein kann. Man könnte denken, die Datei wird per `require` eingebunden. Daher sollte man hier vielleicht noch einen anderen Namen finden.

Reply:
glücklich bin ich damit auch nicht. die lösung ist jetzt nur typ-sicher.. den error verschluckt haben wir vorher schon

Reply:
aber ich sehe die chance bei leerem string noch größer als bei null, dass der fehler noch weiter verschluckt wird.

Reply:
Und für mich sieht die Codestelle jetzt so aus, als wäre hier legitim leerer String vorgesehen.

Reply:
Oder anders gesagt: Mit persönlich ist hier bei der Variante der Preis für die Typsicherheit zu hoch. Da fand ich die Situation vorher, trotz weniger Typsicherheit, besser.

Somit würde ich lieber Status quo lassen, oder Lösung einbauen für Typsicherheit, wo der Fehler gar nicht verschluckt wird (s.o.).